src/main/java/com/google/devtools/build/lib/vfs/DigestUtils.java

```diff
@@ -62,6 +65,7 @@ private static class CacheKey {
     public CacheKey(Path path, FileStatus status) throws IOException {
       this.path = path.asFragment();
       this.nodeId = status.getNodeId();
+      this.changeTime = status.getLastChangeTime();
```

Comment:
This idea seems sound under typical circumstances. I would just be concerned about use cases that play tricks with these timestamps. My general advice would be to not risk it unless there’s a tangible concern (e.g. in this case the increased memory use of the cache key, although if that’s a concern we could optimize for the common case of mtime == ctime by sharing a field).

Reply:
Is this cached used at Google? If so I would be much less worried about this causing an unintentional memory regression. Since the cache key is critical for correctness, I would also prefer to leave it as is.

Reply:
It’s not used under typical circumstances at google since fast digests are available.

Reply:
Anyway, there’s a maximum size configured on the cache which can be adjusted, so I do not think you have to worry at all about the size if the keys, I was just giving an example of why we might want to pursue a riskier change.